src/main/java/org/broadinstitute/hellbender/tools/coveragemodel/SubroutineSignal.java

```diff
@@ -0,0 +1,130 @@
+package org.broadinstitute.hellbender.tools.coveragemodel;
+
+import org.nd4j.linalg.api.ndarray.INDArray;
+
+import java.io.Serializable;
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * This class is used for communicating compound exit signals from computational subroutines.
+ * It is essentially a wrapper around a map.
+ *
+ * @author Mehrtash Babadi &lt;mehrtash@broadinstitute.org&gt;
+ */
+public final class SubroutineSignal implements Serializable {
+
+    private static final long serialVersionUID = -8990591574704241500L;
+
+    private final Map<String, Object> result;
+
+    /**
+     * Private constructor
+     * @param result a key-value map for exit signals
+     */
+    private SubroutineSignal(final Map<String, Object> result) {
+        this.result = result;
+    }
+
+    /**
+     * Fetch a double from exit signals
+     *
+     * @param key string identifier
+     * @return a double
+     */
+    public double getDouble(final String key) {
+        if (result.containsKey(key)) {
```

Comment:
Done. We also have a generic `public <TYPE> TYPE get(final String key)` now for all types as suggested by @samuelklee.